### DISCUSSION TITLE
Allow user to skip days without breaking streak

### DISCUSSION THREAD

**Discussion by anishbajpai014** [iSoron/uhabits#88] [Feature Requests] [answered]
A user should be able to skip a day without breaking the streak if he was unable to perform activity due to external factor. For example, the swimming pool was closed due to a public holiday.

**Comment by andunix** [comment]
It should also be possible to always skip habits on some weekdays. If you can execute a habit only at work (clean desk for example). It should be possible to do this habit five days a week and reacht 100% as fast as other habits.

**Comment by iSoron** [comment]
I think the need of having "skipped days" comes from the fact that streaks are a very poor measure of habit strength. In my opinion, they are inherently flawed. There is ample psychological research showing that a few lapses have negligible impact in habit formation. The formula we currently use to compute strengths does not suffer from this problem. A single lapse after a long streak will have almost zero impact on the score. Only if the lapses become frequent enough, the score will suffer.

This is one reason, by the way, why the app focuses so much on the strength, and not streaks. Until we come up with something to fix the streaks somehow, they will be second-class citizens. One idea would be counting how many days in a row you have maintained a score higher than, say, 90%. In this way, a few lapses would not break the chain.

To summarize, I think the need for this feature comes from an inherent flaw on streaks, so I am not very enthusiastic to add it to the app. Feel free to convince me otherwise.

**Comment by iSoron** [comment]
> It should also be possible to always skip habits on some weekdays

Unless I am missing something, [this is already possible](http://loophabits.org/faq.html#q2).

> It should be possible to do this habit five days a week and reacht 100% as fast as other habits

This is troublesome for habits with very long interval. If we really followed that, after a single repetition of a monthly habit, the score would already jump to 100%. Currently, no matter what the frequency of your habit is, you will need the same number of "perfect repetitions" to reach 100%.

**Comment by anishbajpai014** [comment]
I agree with what you are saying. Streaks probably are a poor measure of habit, but they are easily understandable. While, in the present state of the app, strength seems ambiguous with no documentation about what it signifies.
Which is maybe why a user would be more interested in not breaking a streak. Also, maintaining a long streak provides with a sense of accomplishment which will probably stay absent until he understands what strength actually signifies.

**Comment by torkelsson** [comment]
I agree with @ab014. 

I find it quite motivating trying to keep as long streak as possible, especially those dark, rainy days when you really just want to stay in, instead of go out and do your 10000 steps. After doing the steps you feel so much better both physically and knowing that you have kept the streak for one more day. 

However sometimes it is impossible to meet the goals due to external factors and when you broken the streak, it is som much easier to break it again.

**Comment by abathur** [comment]
@iSoron I just wanted to suggest some alternative interpretations.The need for "skipped days" (along with other requests related to streaks) could be an indication that:
1. The user finds the streak important (perhaps motivating)
2. The user also finds it important to not lie about completing the habit (to preserve the visual/psychological reinforcement of the streak) when some external factor was involved.

I'm not up on the research you're alluding to; my apologies if the studies already account for my next comment. I would imagine that the habit formation studies don't already account for the existence of this application (or a functionally equivalent tool) in the feedback loop and the possibility that some users will be responsive to its feedback cues. At least for the subset of users prone to being either motivated or demotivated by the app, it seems plausible that:
- some users will flourish under a forgiving feedback regime (i.e., a rigid feedback indicator may discourage them or amplify existing discouragement, or they may lack motivation early in a fresh streak when they are so far from their "best" streak), 
- others will flourish under a demanding one (a forgiving feedback indicator may give them the excuse to "cheat" occasionally, or they may find an indicator that doesn't at least match their self-expectations unmotivating).

**Comment by Weygert** [comment]
Interesting discussion about strength vs. streak usage within the app. At least for the first weeks / months I can only set my target based on a streak value, since the strength is far from any value I would consider as a target.

**Comment by c33s** [comment]
for me skip day would be important, loosing the streak beacause of beeing ill ist demotivating

**Comment by openjck** [comment]
Some other apps have this feature. It can be really useful.

For example, a user who works full time might have a habit like *Go for a walk during my lunch break*. What should they do on a holiday? Checking it doesn't make sense, but leaving it unchecked doesn't make much sense either, as it would hurt their score.

**Comment by GeoGriffin** [comment]
This is one reason that got me looking for a different habit tracker, then I realized that I could contribute to the discussion/code... I haven't looked through all the discussions, but has there been consideration for other statuses besides (green) check or (red) x? Maybe the addition of a (blue) dash to indicate a skipped day? Maybe with a configuration option that defaults to the current usage, so that people that like the existing simplicity aren't forced to change how they are using the app.

**Comment by jigneshdarji91** [comment]
> This is one reason, by the way, why the app focuses so much on the strength, and not streaks. Until we come up with something to fix the streaks somehow, they will be second-class citizens. 

I didn't know about `Strength` before reading this. I actually still don't understand what `Score` means. Streaks is a standard construct across a lot of different apps. Duolingo and Todoist are two popular apps that I know use streak as a primary metric for motivation.

I think most people who care about a pause mode or vacation mode find streaks to be a highly motivating factor to continue the habit. There have been multiple days where I forced myself to read even after a party because I didn't want to break the streak. 

There's different tools that work for different people when it comes to motivation. Streaks work for some, while Strength works for the others. We can provide better tools for both. :)

**Comment by real-or-random** [comment]
I fully agree with #88. I'm having this problem right now, and it doesn't feel good. I have a few work-related habits, and I just don't know what to do with these when I'm on vacation or when there's a holiday.

I believe Strength is a very nice concept and what makes this app better than others (at least for me), and I think if your philosophy is Strength (again, depends), then you don't need a feature  to skip days. But you still need to have something which deals with holidays/vacation/being sick. And I think this should be done for habits individually. You may want to do your personal habits but not your work habits. (Still, it's convenient to pause multiple habits at the same time.)

**Comment by iSoron** [comment, marked as answer]
This has been implemented in the `dev` branch (1a05f7d) and should be available in Loop 2.0.
